plugins/serializers/msgpack/metric.go

```diff
@@ -0,0 +1,66 @@
+package msgpack
+
+import (
+	"encoding/binary"
+	"time"
+
+	"github.com/tinylib/msgp/msgp"
+)
+
+//go:generate msgp
+
+// Metric is structure to define MessagePack message format
+// will be used by msgp code generator
+type Metric struct {
+	Name   string                 `msg:"name"`
+	Time   MessagePackTime        `msg:"time,extension"`
+	Tags   map[string]string      `msg:"tags"`
+	Fields map[string]interface{} `msg:"fields"`
+}
+
+// MessagePackTime implements the official timestamp extension type
+// https://github.com/msgpack/msgpack/blob/master/spec.md#timestamp-extension-type
+//
+// tinylib/msgp has been used their own custom extenstion type and the offcial extension
+// is not available. (https://github.com/tinylib/msgp/issues/214)
```

Comment:
So wouldn't it be beneficial for portability if you use a usual type (e.g. nanosecond-timestamp as uint64) instead of a custom-built extension? I suppose in case of the extension the other end also needs to use the same extension!?

Reply:
It is an official extension which is defined in the MessagePack spec. Because this was added later, the tinylib/msgp hasn't implemented it yet. But the other libraries can understand the extension.

Just FYI, negative extension ids are reserved for the predefined extensions.

https://github.com/msgpack/msgpack/blob/master/spec.md#extension-types

Reply:
Understood. So please keep it as is and mark this conversation as resolved.